spec/lib/pretty_text_spec.rb

```diff
@@ -2220,6 +2216,13 @@ def expect_cooked_match(raw, expected_cooked)
 
   it "should strip SCRIPT" do
     expect(PrettyText.cook("<script>alert(42)</script>")).to eq ""
+    expect(PrettyText.cook("<div><script>alert(42)</script></div>")).to eq "<div></div>"
+  end
+
+  it "strips script regardless of sanitize" do
+    expect(
+      PrettyText.cook("<div><script>alert(42)</script></div>", sanitize: false),
+    ).to eq "<div></div>"
```

Comment:
Backfilling for https://github.com/discourse/discourse/blob/acb4c19eeac9ca2053fcdc8155b7532a662a002a/lib/pretty_text.rb#L311